feat: Flink SQL plugin (#16)
confluentinc/cli-plugins: confluent-flink-quickstart/README.md

### [confluent flink quickstart](confluent-flink-quickstart.py)
- Creates Flink compute pool

>     - Displays existing database(s) (cluster) if available

Reviewer: ```suggestion
    - Displays existing databases (Kafka clusters) if available
```

    - Prompts user to select one or create one
    - If no database is found in the region, it will create one

>     - If creating a new cluster, also enables Schema Registry for it 

Reviewer: I think each Schema Registry cluster is tied to a Cloud environment, not a Kafka cluster.
```suggestion
    - If creating a new Kafka cluster, also enable Schema Registry
```

Author: ack - updated the text

- TODO
    - Display all available databases (Kafka cluster) and allow users to interactively pick vs. first one
#### Requirements
- Python 3 (3.10.9 used for this plugin)  `brew install python3`
- [Confluent CLI v3.0.0](https://docs.confluent.io/confluent-cli/current/install.html)
#### Usage
```text
usage: confluent flink quickstart [-h] --name NAME [--units NUM-UNITS] [--env ENV] [--region REGION] 

> Creates Flink compute pool Associates a Kafka cluster for it creating one if none found then starts a Flink SQL cli session This plugin assumes confluent CLI v3.0.0 or greater

Author: Done

options:
  -h, --help            show this help message and exit
  --name NAME           The name for your Flink compute pool

>   --units {5,10}        The number of Confluent Flink Units

Author: I've updated the default. I don't think we should limit users working from the CLI, so I'll leave 10 as an option.

  --env ENV             The environment name

>   --region {us-east-1,us-east-2,eu-central-1,eu-west-1}

Reviewer: You will have to keep updating this list as new regions are added.

Author: Yep that's fine, I want to minimize errors from invalid input

### [confluent flink quickstart](confluent-flink-quickstart.py)
- Creates Flink compute pool

>     - Displays existing databases (Kafka clusters) if available

Reviewer: ```suggestion
    - Displays available databases (Kafka clusters)
```

>     - Prompts user to select one or create one

Reviewer: ```suggestion
    - Prompts user to select one or create a new one
```

>     - If no database is found in the region, it will create one

Reviewer: ```suggestion
    - If no database is found in the region, it will create one automatically
```

>     - If the plugin creates a new Kafka cluster, it also enables Schema Registry

Reviewer: What if it uses an existing Kafka cluster but Schema Registry is not enabled?

Author: Not sure — I can do a follow-up PR that does a check for SR being enabled — but I don't know if that's necessary, i.e I would imagine creating a Flink pool automatically enables SR.

Reviewer: I'm not sure if that's the case, but in Q1 of next year it looks like SR will be enabled for all users by default so if there's an issue here it'll fix itself

> - Creates a Flink shell session once the pool is running

Reviewer: ```suggestion
- Starts a Flink shell session once the Flink compute pool is running
```

- 
#### Requirements
- Python 3 (3.10.9 used for this plugin)  `brew install python3`

> - [Confluent CLI v3.0.0](https://docs.confluent.io/confluent-cli/current/install.html)

Reviewer: One of the latest minor versions would be preferred, since the Flink commands weren't introduced until recently.
```suggestion
- [Confluent CLI v3](https://docs.confluent.io/confluent-cli/current/install.html)
```

#### Usage
```text
usage: confluent flink quickstart [-h] --name NAME [--units NUM-UNITS] [--env ENV] [--region REGION] 

> This plugin assumes confluent CLI v3.0.0 or greater

Reviewer: These Flink commands are still in EA and subject to breaking changes, so we should probably stay safe and say something to the effect of "This plugin assumes you have installed the latest Confluent CLI v3"

options:
  -h, --help            show this help message and exit
  --name NAME           The name for your Flink compute pool
  --units {5,10}        The number of Confluent Flink Units
  --env ENV             The environment name
  --region {us-east-1,us-east-2,eu-central-1,eu-west-1}
                        Cloud region defaults to us-east-1
  --cloud {aws}         Cloud defaults to aws

>   --debug {y,n}         Prints the results of every command, defaults to n

Reviewer: Let's make this a boolean flag.
```suggestion
  --debug               Prints the results of every command, defaults to false
```

### [confluent flink quickstart](confluent-flink-quickstart.py)
- Creates Flink compute pool
    - Displays existing database(s) (cluster) if available
    - Prompts user to select one or create one
    - If no database is found in the region, it will create one
    - If creating a new cluster, also enables Schema Registry for it 
- Creates a Flink shell session once the pool is running
- 
#### Requirements
- Python 3 (3.10.9 used for this plugin)  `brew install python3`
- [Confluent CLI v3.0.0](https://docs.confluent.io/confluent-cli/current/install.html)
#### Usage
```text
usage: confluent flink quickstart [-h] --name NAME [--units NUM-UNITS] [--env ENV] [--region REGION] 

> Creates a Flink compute pool.

Reviewer: ```suggestion
Create a Flink compute pool.
```

Author: I think since it's referring to a singular noun, it should be `Creates`

Reviewer: I just wanted to match the description here: https://docs.confluent.io/confluent-cli/current/command-reference/flink/compute-pool/confluent_flink_compute-pool_create.html#confluent-flink-compute-pool-create

Author: I'll update to match it then

> Looks for exising Kafka clusters and prompts the user to select one as the database for the Flink pool.

Reviewer: ```suggestion
Look for existing Kafka clusters and prompt the user to select one as the database for the Flink compute pool.
```

Creating one is an option as well.                                           

> If there are no existing clusters, the plugin will automatically create one.

Reviewer: ```suggestion
If there are no existing Kafka clusters, the plugin will automatically create one.
```

> Then it starts a Flink SQL cli session

Reviewer: ```suggestion
Then it starts a Flink SQL CLI session.
```

This plugin assumes confluent CLI v3.0.0 or greater

options:
  -h, --help            show this help message and exit
  --name NAME           The name for your Flink compute pool

>   --units {5,10}        The number of Confluent Flink Units

Reviewer 2: maybe `--cfu` instead of `--units`. would be analogous to `confluent kafka cluster create my-cluster` taking `--cku int`

Reviewer: Was this updated? I still see the old flag name.

Author: It is now

>   --env ENV             The environment name

Reviewer 2: suggest `--environment` to be consistent with other CLI commands (`confluent kafka cluster create`, confluent flink compute-pool create)
